test/Flows.test.ts

```diff
@@ -155,29 +154,33 @@ options.forEach(params => {
 
     if (params.relay) {
       let approvalPaymaster: TestPaymasterPreconfiguredApprovalInstance
+      let relayProvider: RelayProvider
 
       describe('request with approvaldata', () => {
-        let approvalData: PrefixedHexString
         before(async function () {
           approvalPaymaster = await TestPaymasterPreconfiguredApproval.new()
           await approvalPaymaster.setRelayHub(rhub.address)
           await approvalPaymaster.setTrustedForwarder(await sr.getTrustedForwarder())
           await rhub.depositFor(approvalPaymaster.address, { value: (1e18).toString() })
+        })
 
-          const relayProvider =
+        const setRecipientProvider = function (asyncApprovalData: AsyncDataCallback): void {
+          relayProvider =
             // @ts-ignore
             new RelayProvider(web3.currentProvider,
-              relayClientConfig, { asyncApprovalData: async () => await Promise.resolve(approvalData) })
+              relayClientConfig, { asyncApprovalData })
           TestRecipient.web3.setProvider(relayProvider)
-        })
+        }
```

Comment:
This is a weird construct tbh

Reply:
I wanted a local function to set the provider with "custom" param.
it didn't let me define a function in this scope, only variables - and it does have side-effect of modifying "describe"-context variable, so it can't be moved out